valhalla/sif/dynamiccost.h

```diff
@@ -605,7 +608,7 @@ class DynamicCost {
         if (access_type == baldr::AccessType::kTimedAllowed)
           time_allowed = true;
 
-        if (current_time == 0) {
+        if (current_time == 0 || ignore_non_vehicular_restrictions_) {
```

Comment:
I'd put this into the `if` on line 602, so `time_allowed` will stay false if this option is given and we don't need it in the `return` below.